chia/data_layer/data_layer.py

```diff
@@ -117,11 +118,12 @@ async def batch_update(
                 key = change["key"]
                 await self.data_store.delete(key, tree_id, hint_keys_values)
 
-        await self.data_store.get_tree_root(tree_id)
-        root = await self.data_store.get_tree_root(tree_id)
+        new_root = await self.data_store.get_tree_root(tree_id)
+        if old_root == new_root:  # does this comparison work?
```

Comment:
I think this was supposed to change to checking the hash?  We wouldn't want status or generation to affect this check.

```suggestion
        if old_root.node_hash == new_root.node_hash:
```

Reply:
changed and updated